### ISSUE TITLE
BassFX doesn't work under Linux

### ISSUE THREAD

**Issue by recapitalverb** [Wieku/danser#7]
It looks like libbass is missing when I try to run it using the `./gradlew desktop:run` command as specified in README.md. How should I work around this?
Full logs:
```
reverb ~/Repos/danser ➜ ./gradlew desktop:run

> Task :desktop:run FAILED
Version 20.0112-48daff9
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f3f144bbeb7, pid=29815, tid=0x00007f3f4040a700
#
# JRE version: OpenJDK Runtime Environment (8.0_232-b09) (build 1.8.0_232-b09)
# Java VM: OpenJDK 64-Bit Server VM (25.232-b09 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libbass_fx.so+0xeeb7]
#
# Core dump written. Default location: /home/reverb/Repos/danser/desktop/run/core or core.29815
#
# An error report file with more information is saved as:
# /home/reverb/Repos/danser/desktop/run/hs_err_pid29815.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':desktop:run'.
> Process 'command '/usr/lib/jvm/java-8-openjdk/bin/java'' finished with non-zero exit value 134

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
21 actionable tasks: 1 executed, 20 up-to-date
```

**Comment by Wieku** [comment]
Hmm, it looks like the lib is loaded. I think it is an issue in [NativeBass](https://github.com/jerome-jouvie/NativeBass) or bug in bassfx for linux itself. Will look into that later.

**Comment by Wizkiller96** [comment]
Danser runs for me but Danser! button noting happen just show a blank screen.

**Comment by Wieku** [comment]
@Wizkiller96 what do you mean? Screenshots, logs, anything?

**Comment by Wizkiller96** [comment]
![image](https://user-images.githubusercontent.com/3111165/73478628-6fd14e00-4364-11ea-822f-e70ade076c5a.png)

**Comment by Wieku** [comment]
So it's working as it should be. It's a fresh project, I even don't have hit objects implemented yet

**Comment by Wizkiller96** [comment]
Then it all good then though I did something wrong. Thanks for the quick response.

**Comment by Icaro1447** [comment]
same problem for me: https://imgur.com/MnqPdnF

**Comment by Wieku** [comment]
@Icaro1447 no, you just can't read the messages because you don't have JDK. @recapitalverb issue is different.

**Comment by recapitalverb** [comment]
@Icaro1447 You're missing JDK, install it.

**Comment by recapitalverb** [comment]
On the issue: I have not tried running this with Oracle's JDK instead of OpenJDK. Will try when possible, though I doubt that we can just say "it runs on Oracle's JDK, issue solved".

**Comment by developomp** [comment]
Just ran it with Oracle JDK jdk1.8.0_251
It still doesn't want to run

```
pomp@pomp:~/Downloads/danser$ ./gradlew desktop:run

> Task :desktop:run FAILED
2020-05-27 09:52:32.066 [main] [Runtime] [INFO]: Window mode changed: Windowed -> Windowed
2020-05-27 09:52:32.068 [main] [Runtime] [INFO]: Window mode changed: Windowed -> Windowed
/usr/lib/jvm/jdk1.8.0_251/bin/java: symbol lookup error: /tmp/lwjglpomp/3.2.3-build-13/libbass_fx.so: undefined symbol: BASS_GetVersion

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':desktop:run'.
> Process 'command '/usr/lib/jvm/jdk1.8.0_251/bin/java'' finished with non-zero exit value 127

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
21 actionable tasks: 1 executed, 20 up-to-date
pomp@pomp:~/Downloads/danser$ 
```

**Comment by developomp** [comment]
I also ran it in Windows just in case but am still getting an error
```
C:\Users\pomp\Downloads\danser>gradlew desktop:run

> Task :desktop:run FAILED
[LWJGL] GLFW_VERSION_UNAVAILABLE error
        Description : WGL: Failed to create OpenGL context
        Stacktrace  :
                org.lwjgl.glfw.GLFW.nglfwCreateWindow(GLFW.java:1714)
                org.lwjgl.glfw.GLFW.glfwCreateWindow(GLFW.java:1897)
                me.wieku.framework.backend.DesktopContext.startContext(DesktopContext.kt:152)
                me.wieku.framework.backend.GameContext.start(GameContext.kt:59)
                me.wieku.danser.MainKt.main(Main.kt:18)
2020-05-27 10:39:11.874 [main] [Runtime] [INFO]: Window mode changed: Windowed -> Windowed
Exception in thread "main" java.lang.NullPointerException
        at org.lwjgl.system.Checks.check(Checks.java:100)
        at org.lwjgl.glfw.GLFW.glfwSetWindowAttrib(GLFW.java:2726)
        at me.wieku.framework.backend.DesktopContext.setWindowMode(DesktopContext.kt:43)
        at me.wieku.framework.backend.DesktopContext.startContext(DesktopContext.kt:153)
        at me.wieku.framework.backend.GameContext.start(GameContext.kt:59)
        at me.wieku.danser.MainKt.main(Main.kt:18)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':desktop:run'.
> Process 'command 'C:\Program Files\Java\jdk1.8.0_251\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
21 actionable tasks: 1 executed, 20 up-to-date

C:\Users\pomp\Downloads\danser> 
```

**Comment by Wieku** [comment]
> I also ran it in Windows just in case but am still getting an error
> 
> ```
> C:\Users\pomp\Downloads\danser>gradlew desktop:run
> 
> > Task :desktop:run FAILED
> [LWJGL] GLFW_VERSION_UNAVAILABLE error
>         Description : WGL: Failed to create OpenGL context
>         Stacktrace  :
>                 org.lwjgl.glfw.GLFW.nglfwCreateWindow(GLFW.java:1714)
>                 org.lwjgl.glfw.GLFW.glfwCreateWindow(GLFW.java:1897)
>                 me.wieku.framework.backend.DesktopContext.startContext(DesktopContext.kt:152)
>                 me.wieku.framework.backend.GameContext.start(GameContext.kt:59)
>                 me.wieku.danser.MainKt.main(Main.kt:18)
> 2020-05-27 10:39:11.874 [main] [Runtime] [INFO]: Window mode changed: Windowed -> Windowed
> Exception in thread "main" java.lang.NullPointerException
>         at org.lwjgl.system.Checks.check(Checks.java:100)
>         at org.lwjgl.glfw.GLFW.glfwSetWindowAttrib(GLFW.java:2726)
>         at me.wieku.framework.backend.DesktopContext.setWindowMode(DesktopContext.kt:43)
>         at me.wieku.framework.backend.DesktopContext.startContext(DesktopContext.kt:153)
>         at me.wieku.framework.backend.GameContext.start(GameContext.kt:59)
>         at me.wieku.danser.MainKt.main(Main.kt:18)
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':desktop:run'.
> > Process 'command 'C:\Program Files\Java\jdk1.8.0_251\bin\java.exe'' finished with non-zero exit value 1
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 2s
> 21 actionable tasks: 1 executed, 20 up-to-date
> 
> C:\Users\pomp\Downloads\danser> 
> ```

This issue is unrelated. It seems that you are missing OpenGL drivers.

**Comment by Wieku** [comment]
Fixed by: 027c8e8
